dspace-api/src/main/java/org/dspace/app/util/AuthorizeUtil.java

```diff
@@ -410,15 +414,15 @@ else if (!AuthorizeManager.isAdmin(context))
     public static void authorizeRemoveAdminGroup(Context context,
             Collection collection) throws AuthorizeException, SQLException
     {
-        Community[] parentCommunities = collection.getCommunities();
+        List<Community> parentCommunities = collection.getCommunities();
         if (AuthorizeConfiguration
                 .canCommunityAdminManageCollectionAdminGroup()
-                && parentCommunities != null && parentCommunities.length > 0)
+                && parentCommunities != null && parentCommunities.size() > 0)
         {
-            AuthorizeManager.authorizeAction(context, collection
-                    .getCommunities()[0], Constants.ADMIN);
+            authorizeService.authorizeAction(context, collection
+                    .getCommunities().get(0), Constants.ADMIN);
```

Comment:
@peterdietz This is the correct way, you can get the communities from from a collection by using collection.getCommunities(). This is a database mapping that is a simple getter. These can still be made from the original database object.

Hope this helps.